Popup/Popup.js

```diff
@@ -653,36 +403,275 @@ class Popup extends Component {
 			if (current) {
 				current.blur();
 			}
-			Spotlight.setActiveContainer(containerId);
+			Spotlight.setActiveContainer(containerId.current);
 		}
-	};
+	}, [open]);
+
+	const getDerivedStateFromProps = useCallback(() => {
+		if (open !== prevOpen) {
+			if (open) {
+				setPopupOpen(noAnimation || floatLayerOpen ? OpenState.OPEN : OpenState.CLOSED);
+				setFloatLayerOpen(true);
+				setActivator(Spotlight.getCurrent());
+				setPrevOpen(open);
+			} else {
+				// Disables the spotlight conatiner of popup when `noAnimation` set
+				if (noAnimation) {
+					const node = getContainerNode(containerId.current);
+					if (node) {
+						node.dataset['spotlightContainerDisabled'] = true;
+					}
+				}
 
-	render () {
-		const {noAutoDismiss, scrimType, ...rest} = this.props;
+				setPopupOpen(OpenState.CLOSED);
+				setFloatLayerOpen(popupOpen === OpenState.OPEN ? !noAnimation : false);
+				setActivator(noAnimation ? null : activator);
+				setPrevOpen(open);
+			}
+		}
+	}, [activator, floatLayerOpen, noAnimation, open, popupOpen, prevOpen]);
+
+	const handleComponentUpdate = useCallback(() => {
+		if (open !== prevOpen) {
+			if (!noAnimation) {
+				if (!open && popupOpen === OpenState.OPENING || !open && popupOpen === OpenState.OPEN) {
```

Comment:
can you explain why this changed from using `popupOpen === OpenState.CLOSED`  ?

Reply:
with the current implementation and async state change of the `popupOpen`, if we use `popupOpen === OpenState.CLOSED`, the spotlight navigation will not be resumed when the popup is open and closed quickly